### PR TITLE
Core/ItemHandler: fix selling items over gold limit

### PR DESCRIPTION
**Pull request by okfok**
Changes proposed:

Make it impossible to sell items if it will cause exceeding gold limit

Issues:

when you sell items over gold limit that item will be sold but you wont get money for it.

Tests performed:

Build successful, 
tested in-game

### CODE REVIEW

File: src/server/game/Handlers/ItemHandler.cpp [hunk 1]
```diff
@@ -440,6 +440,13 @@ void WorldSession::HandleSellItemOpcode(WorldPacket& recvData)
         ItemTemplate const* pProto = pItem->GetTemplate();
         if (pProto)
         {
+            uint32 money = pProto->SellPrice * count;
+            if (_player->GetMoney() + money > MAX_MONEY_AMOUNT)               // prevent exceeding gold limit
+            {
+                _player->SendEquipError(EQUIP_ERR_TOO_MUCH_GOLD, nullptr, nullptr);
+                _player->SendSellError(SELL_ERR_UNK, creature, itemguid, 0);
+                return;
+            }
```
Comment (jackpoz) [inline src/server/game/Handlers/ItemHandler.cpp:449]: this code should be in the `if (pProto->SellPrice > 0)` body.

File: src/server/game/Handlers/ItemHandler.cpp [hunk 2]
```diff
@@ -440,6 +440,13 @@ void WorldSession::HandleSellItemOpcode(WorldPacket& recvData)
         ItemTemplate const* pProto = pItem->GetTemplate();
         if (pProto)
         {
+            uint32 money = pProto->SellPrice * count;
+            if (_player->GetMoney() + money > MAX_MONEY_AMOUNT)               // prevent exceeding gold limit
```
Comment (jackpoz) [inline src/server/game/Handlers/ItemHandler.cpp:444]: Check similar code like https://github.com/TrinityCore/TrinityCore/blob/0d8672732552b0474e22b077a2318fa87446f25f/src/server/game/Handlers/TradeHandler.cpp#L299 to make sure `_player->GetMoney() + money` doesn't overflow (and please test it with the overflow case to verify it works as expected)

File: src/server/game/Handlers/ItemHandler.cpp (hunk 1 above)
Comment (okfok) [reply src/server/game/Handlers/ItemHandler.cpp:449]: ok, i will move it

File: src/server/game/Handlers/ItemHandler.cpp (hunk 2 above)
Comment (okfok) [reply src/server/game/Handlers/ItemHandler.cpp:444]: yep, it has trouble when item costs 1 copper

Comment (okfok) [reply src/server/game/Handlers/ItemHandler.cpp]: fixed

**Review by jackpoz — APPROVED** [review]

**Comment by jackpoz** [comment]
Thanks for the PR 🚀 😃

**Pull request merged**